fix: do not skip po file generation on missing sources

regisb opened the pull request:
Previously, when some po files were missing, the django.po or
djangojs.po were not generated at all. This was very inconvenient for
some languages (e.g: 'ro') that do not pass the 3% bar of reviewed
strings in some files. Here, we make a change where the remaining po
files are merged to produce the merged.po files.

This fixes an issue that occurs when we generate translation strings for the Romanian language in Tutor. See conversation here: https://discuss.overhang.io/t/translate-tutor-in-other-language/1078

This is ready to review.

edx-webhook commented:
Thanks for the pull request, @regisb! I've created [OSPR-5621](https://openedx.atlassian.net/browse/OSPR-5621) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

natabene commented:
@regisb Thank you for your contribution. @kashifch Could someone from your team please review?

DawoudSheraz reviewed (commented):
Can you add related unit tests? Thank you

regisb commented:
> Can you add related unit tests? Thank you

@DawoudSheraz Does this mean that the implementation change is acceptable? I don't want to spend time on writing unit tests if the new implementation has to be amended.

DawoudSheraz commented:
> > Can you add related unit tests? Thank you
> 
> @DawoudSheraz Does this mean that the implementation change is acceptable? I don't want to spend time on writing unit tests if the new implementation has to be amended.

Yes, the changes look good.

regisb commented:
Thanks for the quick response @DawoudSheraz. I added unit tests.

DawoudSheraz commented on tests/test_generate.py:
```diff
@@ -45,6 +45,38 @@ def test_merge(self):
+
+    def test_merge_with_missing_sources_strict(self):
+        """
+        Tests merge script on English source files.
```
the docstring is not descriptive.

DawoudSheraz reviewed (approved):
A nit comment. Once addressed, it will be good to merge.

awaisdar001 commented on i18n/generate.py:
```diff
@@ -135,19 +139,18 @@ def clean_pofile(pofile_path):
-        clean_pofile(pathname)
+    pathname = directory.joinpath(filename)
+    if not pathname.exists():
+        raise ValueError(u"I18N: Cannot generate because file not found: {0}".format(pathname))
```
Do we need this unicode `u" ... "` as we have migrated to python3+?

regisb replied:
No we don't -- this comes from a difficult rebase.

regisb commented:
Thanks for the review @DawoudSheraz. I addressed your comments -- and took the liberty to also address minor linting issues. I know that this is post review, but I hope that's ok?

edx-webhook commented:
@regisb 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/i18n-tools&entry.867055334=https://github.com/edx/i18n-tools/pull/95&entry.1484655318=https://github.com/regisb&entry.752974735=2021-02-09+22:25&entry.1917517419=2021-02-17+11:15&entry.2133058324=Yes) so we can improve your experience in the future.

DawoudSheraz commented:
@regisb Thank you for this contribution.

regisb commented:
Thanks for the quick review @DawoudSheraz!
